xbmc/video/VideoDatabase.cpp

```diff
@@ -11899,61 +11898,55 @@ void CVideoDatabase::GetVideoVersions(VideoDbContentType itemType,
   }
 }
 
-void CVideoDatabase::GetDefaultVideoVersion(VideoDbContentType itemType, int dbId, CFileItem& item)
+int CVideoDatabase::GetDefaultVideoVersionId(VideoDbContentType itemType, int dbId)
 {
   if (!m_pDB || !m_pDS)
-    return;
+    return -1;
 
   MediaType mediaType;
-  std::string strSQL;
+  std::string query;
 
   if (itemType == VideoDbContentType::MOVIES)
   {
     mediaType = MediaTypeMovie;
-    strSQL = PrepareSQL("SELECT videoversiontype.name AS name,"
-                        "  videoversiontype.id AS id,"
-                        "  videoversion.idFile AS idFile "
-                        "FROM videoversiontype"
-                        "  JOIN videoversion ON"
-                        "    videoversion.idType = videoversiontype.id"
-                        "  JOIN movie ON"
-                        "    movie.idFile = videoversion.idFile "
-                        "WHERE movie.idMovie = %i",
-                        dbId);
+    query = PrepareSQL("SELECT idFile FROM movie WHERE movie.idMovie = %i", dbId);
```

Comment:
I think the whole method can be implemented much more simple using `CDatabase::GetSingleValueInt`.

Reply:
CDatabase::GetSingleValueInt returns 0 when failed, we need -1.

Reply:
I don't think it's viable to keep adding a new function and SQL query every time we need something slightly different.
The movie table already has helper functions. `GetDetailsByTypeAndId()` should be able to do the job?

Right now `GetDetailsByTypeAndId()` doesn't have a parameter for the bit field getDetails to avoid unnecessary additional queries, but the functions it calls already accept it (see `GetMovieInfo()` etc.) Please add that parameter, and this function can send a 0 as bit field value because all it needs is in the movie table. That part should be 1 dedicated commit that comes before the commit that contains your fix.